vividus-extension-selenium/src/main/java/org/vividus/bdd/proxy/ProxyAgentStoryReporter.java

```diff
@@ -39,15 +40,10 @@
     public void beforeStory(Story story, boolean givenStory)
     {
         String name = bddRunContext.getRunningStory().getName();
-        if (!configuration.dryRun())
+        if (!configuration.dryRun() && !givenStory && !"BeforeStories".equals(name) && !"AfterStories".equals(name)
```

Comment:
Does it make sense to refactor this complex condition to something like:
`if (!configuration.dryRun() && !givenStory && notPrecodition(name) && proxyEnabled()) `